pyproject.toml

```diff
@@ -126,7 +123,8 @@ tests = [
     "pympler~=0.9",
     "coverage<5.0",
     "sqlalchemy-utils~=0.37.2",
-    "sphinx~=3.2.1"
+    "sphinx~=4.0",
```

Comment:
Any reason to have `~=4.0` here but `~=4.1` in the `docs` extra?

Reply:
`nitpick_ignore_regex` was only introduced in 4.1, added for our docs build, but for testing the aiida sphinx extension, it is only sphinx 4 that is needed (it still actually works for sphinx 3, but there is a change in the AST, when regression testing the output)